demoTests/SafetyScreenTestCase.swift

```diff
@@ -0,0 +1,188 @@
+        interactor.visionManager(VisionManager.shared, didUpdateWorldDescription: worldDescription)
+        XCTAssert(presenter.currentSafetyState == .none, "Distance to lead sign shouldn't be presented")
+    }
+    
+    func testChaningState() {
+        
+        let worldDescription = WorldDescription([(.car, .warning)])
+        interactor.visionManager(VisionManager.shared, didUpdateWorldDescription: worldDescription)
+        let result = SafetyState.collisions([.warning(.car(WorldDescription.bbox))])
+        XCTAssert(presenter.currentSafetyState == result, "Collision Warning should be presented")
+        
+        let anotherWorldDescription = WorldDescription([(.car, .notTriggered)])
+        interactor.visionManager(VisionManager.shared, didUpdateWorldDescription: anotherWorldDescription)
+        XCTAssert(presenter.currentSafetyState == distanceState, "Distance to lead car should be presented")
+    }
+    
+    func testForwarCar() {
+        
+        let worldDescription = WorldDescription([(.person, .notTriggered), (.car, .notTriggered)], forwardCarIndex: 1)
+        interactor.visionManager(VisionManager.shared, didUpdateWorldDescription: worldDescription)
+        XCTAssert(presenter.currentSafetyState == distanceState, "Distance to lead car should be presented")
+    }
+}
+
+extension WorldDescription {
+    
+    static let distance: Double = 1.0
+    
+    static let bbox: CGRect = .zero
+    static let coords = WorldCoordinate(x: 0, y: 0, z: 0)
+    
+    convenience init(_ params: [(ObjectType, CollisionState)], forwardCarIndex: UInt = 0) {
+        
+        var objects: [ObjectDescription] = []
+        var collisionObjects: [CollisionObjectDescription] = []
+        
+        for param in params {
+            let detection = Detection(identifier: Identifier(),
+                                      boundingBox: WorldDescription.bbox,
+                                      objectType: param.0,
+                                      confidence: 0)
+            
+            let object = ObjectDescription(identifier: Identifier(),
+                                          distance: WorldDescription.distance,
+                                          worldCoordinate: WorldDescription.coords,
+                                          detection: detection)
+            objects.append(object)
+            
+            let collision = CollisionObjectDescription(object: object,
+                                                       timeToImpact: 0,
+                                                       state: param.1)
+            collisionObjects.append(collision)
+        }
+        
+        self.init(identifier: Identifier(),
+                  forwardCarIndex: NSNumber(value: forwardCarIndex),
+                  objects: objects,
+                  collisionObjects: collisionObjects)
+    }
+}
+
+class MockContainerPresenrer: ContainerPresenter {
```

Comment:
Probably may be needed not only in this test file. Please consider extracting to separate file.